### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ratranqu**

### CODE REVIEW

File: Sources/Tendermint/RESTClient/RESTResponses.swift [hunk 1]
```diff
@@ -90,6 +90,11 @@ public struct CheckTransactionResponse: Codable {
 //    type ResultCheckTx struct {
 //        abci.ResponseCheckTx
 //    }
+    let code: UInt32
+    let data: Data // bytes.HexBytes
+    let log: String
+    let codespace: String // TODO: is this necessary?
+    let hash: Data //bytes.HexBytes
 }
```
Comment (JaapWijnen) [inline Sources/Tendermint/RESTClient/RESTResponses.swift:94]: abci has the following type for ResponseCheckTx so it seems there's some properties missing still?

```
   public let code: UInt32
    /// Result bytes, if any.
    public let data: Data
    /// The output of the application's logger. May be non-deterministic.
    public let log: String
    /// Additional information. May be non-deterministic.
    public let info: String
    /// Amount of gas required for the transaction.
    public let gasWanted: Int64
    /// Amount of gas consumed by the transaction.
    public let gasUsed: Int64
    /// Type and Key-Value events for indexing transactions (eg. by account).
    public let events: [Event]
    /// Namespace for the `code`.
    public let codespace: String
```

Comment (JaapWijnen) [reply Sources/Tendermint/RESTClient/RESTResponses.swift:94]: mainly `info`, `gasWanted`, `gasUsed` and `events`
and there's no hash on the responsechecktx type. So not sure where this is coming from?

File: Sources/Tendermint/Types/Transaction.swift [hunk 2]
```diff
@@ -6,3 +6,16 @@
 //
 
 import Foundation
+
+
+public typealias TransactionBytes = Data
+
+extension TransactionBytes {
+    var data: Data {
```
Comment (JaapWijnen) [inline Sources/Tendermint/Types/Transaction.swift]: why add a computed property of data? since you can just use self and have the exact same api no?

**Review by paulofaria — APPROVED** [review]
LGTM

**Pull request merged**